run/logging-manual/test/system.test.js

```diff
@@ -0,0 +1,114 @@
+const assert = require('assert');
+const {Logging} = require('@google-cloud/logging');
+const request = require('got');
+
+const promisify = require('util').promisify;
+const setTimeoutPromise = promisify(setTimeout);
+
+// Support concurrency by setting the service name to something unique.
+// The service name must be the same used to deploy to Cloud Run.
+const service_name = process.env.CLOUD_RUN_SERVICE_NAME || require('../package').name;
+
+const logging = new Logging({
+  projectId: process.env.GOOGLE_CLOUD_PROJECT
+});
+
+const getLogEntriesPolling = async (filter, max_attempts) => {
+  const WRITE_CONSISTENCY_DELAY_MS = 10000;
+  const MAX_ATTEMPTS = max_attempts || 8;
+  let entries;
+
+  for (i = 0; i < MAX_ATTEMPTS; i++) {
+    await setTimeoutPromise(WRITE_CONSISTENCY_DELAY_MS);
+    entries = await getLogEntries(filter);
+    if (entries[0] && entries[0].length > 0) {
+      return entries[0];
+    }
+  }
+
+  return [];
+};
+
+const getLogEntries = async (filter) => {
+  const preparedFilter = `resource.type="cloud_run_revision" ${filter}`;
+  let entries = await logging.getEntries({
+    filter: preparedFilter,
+    autoPaginate: false,
+    pageSize: 2
+  });
+
+  return entries;
+}
+
+const dateMinutesAgo = (date, min_ago) => {
+  date.setMinutes(date.getMinutes() - min_ago)
+  return date.toISOString();
+}
+
+describe('Logging', () => {
+
+  let requestLog;
+  let sampleLog;
+
+  describe("Live Service", () => {
+    it ('can be reached by an HTTP request', async () => {
+      const url = process.env.CLOUD_RUN_SERVICE_URL;
+      if (!url) {
+        throw Error('"$CLOUD_RUN_SERVICE_URL" environment variable is required. For example: https://service-x8xabcdefg-uc.a.run.app.')
+      }
+      await request(url);
+    });
+
+    it ('generates Stackdriver logs', async() => {
+      // The latest two log entries for our service in the last 5 minutes
+      // are treated as the result of the previous test.
+      // Concurrency is supporting by distinctly named service deployment per test run.
+      let filter = `resource.labels.service_name="${service_name}" timestamp>="${dateMinutesAgo((new Date), 5)}"`;
+      const entries = await getLogEntriesPolling(filter);
+      entries.forEach(entry => {
+        if (entry.metadata.httpRequest) {
+          requestLog = entry;
```

Comment:
Nit: are these values used for anything within this `describe` block?

(I suspect they _may_ be used in the _subsequent_ one, but personally I feel like that's bad practice. @fhinkel WDYT?)

Reply:
Correct, the variables they are assigning to are in the top-level describe but for use in subsequent describes. It is a questionable practice.

These started out by being in before() statements, but I decided that checking the service is live and Stackdriver has logs at all should be test cases in their own right.